avalon/tools/cbsceneinventory/app.py

```diff
@@ -224,6 +239,199 @@ def show_remove_warning_dialog(self, items):
         self.data_changed.emit()
 
 
+class SwitchAssetDialog(QtWidgets.QDialog):
+    """Widget to support asset switching"""
+
+    switched = QtCore.Signal()
+
+    def __init__(self, parent=None, items=None):
+        QtWidgets.QDialog.__init__(self, parent)
+
+        self._items = items
+
+        self._assets_box = None
+        self._subsets_box = None
+        self._representations_box = None
+
+        input_layout = QtWidgets.QHBoxLayout()
+        input_layout.setDirection(QtWidgets.QBoxLayout.RightToLeft)
+
+        accept_icon = qta.icon("fa.check-square", color="white")
+        accept_btn = QtWidgets.QPushButton()
+        accept_btn.setIcon(accept_icon)
+        accept_btn.setFixedWidth(24)
+        accept_btn.setFixedHeight(24)
+
+        input_layout.addWidget(accept_btn)
+
+        self._input_layout = input_layout
+        self._accept_btn = accept_btn
+
+        self.setLayout(input_layout)
+        self.setWindowTitle("Switch selected items ...")
+
+        self.connections()
+
+        self.refresh()
+
+    def connections(self):
+        self._accept_btn.clicked.connect(self._on_accept)
+
+    def refresh(self):
+
+        assets = self._get_assets()
+        self._assets_box = self._create_combo_box(assets)
+
+        subsets = self._get_subsets()
+        self._subsets_box = self._create_combo_box(subsets)
+
+        representations = self._get_representations()
+        self._representations_box = self._create_combo_box(representations)
+
+        self._input_layout.addWidget(self._representations_box)
+        self._input_layout.addWidget(self._subsets_box)
+        self._input_layout.addWidget(self._assets_box)
+
+    def _create_combo_box(self, items):
+        """Create a combobox with auto completion, first item will be "----"
+
+        Args:
+            items (iterable): list of document names
+
+        Returns:
+            QtWidgets.QComboBox
+
+        """
+
+        combobox = QtWidgets.QComboBox()
+        completer = QtWidgets.QCompleter(items)
```

Comment:
This is actually pretty interesting that this is how a ComboBox works, haha. Why would it need a separate autocompleter if on line 311 it actually just has the content - why does it need two references to the same data? Isn't it just an autocompletion for the QComboBox... which already contains the data. Oh, oh... Qt. :see_no_evil: Probably an Abstraction level because it's also used for other editors like QLineEdit.